backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Core/Services/ResourceAdministrationPoint.cs

```diff
@@ -154,23 +177,110 @@ private async Task<List<ServiceResource>> GetResources()
+            }
+
+            return resources;
+        }
+
+        /// <summary>
+        /// Filters the provided list of resources based on provided resourceOwnerFilters.
+        /// <param name="resources">The list of resources to be filtered.</param>
+        /// <param name="resourceOwnerFilters">The list of resource owners to be included in the returned list.</param>
+        /// </summary>
+        /// <returns>List of filtered resources or all resources if the list of filters is null or empty</returns>
+        private List<ServiceResourceFE> FilterResourceList(List<ServiceResourceFE> resources, string[] resourceOwnerFilters)
+        {
+            if (resourceOwnerFilters.IsNullOrEmpty())
+            {
+                return resources;
+            }
+
+            List<ServiceResourceFE> filteredResources = new List<ServiceResourceFE>();
+            string[] lowercaseResourceOwnerFilters = Array.ConvertAll(resourceOwnerFilters, d => d.ToLower());
+
+            foreach (ServiceResourceFE res in resources)
+            {
+                if (lowercaseResourceOwnerFilters.Contains(res.ResourceOwnerName.ToLower()))
+                {
+                    filteredResources.Add(res);
+                }
+            }
+
+            return filteredResources;
+        }
+
+        /// <summary>
+        /// Searches through a list of resources after occurences of the words in a search string, resturning a list of resources where at least one of these words are present.
+        /// <param name="resources">The list of resources to be searched through.</param>
+        /// <param name="searchString">The search string to be used for the search.</param>
+        /// </summary>
+        /// <returns>List of resources containing at least one word from the search string, ordered by the number of word occurences found. Returns the full list of resources if the search string is null or empty.</returns>
+        private List<ServiceResourceFE> SearchInResourceList(List<ServiceResourceFE> resources, string? searchString)
+        {
+            if (searchString.IsNullOrEmpty())
+            {
+                return resources;
+            }
+
+            List<SortableServiceResourceFE> matchedResources = new List<SortableServiceResourceFE>();
+            string[] searchWords = searchString.ToLower().Split();
+
+            foreach (ServiceResourceFE res in resources)
+            {
+                int numMatches = 0;
+
+                foreach (string word in searchWords)
+                {
+                    if (res.Title.ToLower().Contains(word) || res.Description.ToLower().Contains(word) || res.RightDescription.ToLower().Contains(word))
+                    {
+                        numMatches++;
+                    }
+                }
+
+                if (numMatches > 0)
+                {
+                    matchedResources.Add(new SortableServiceResourceFE(res, numMatches));
```

Comment:
Huh, I guess we could..
I mean, my thought was that I didn't want to mess with an already established and much used model, but if I just make the property nullable I guess it will just go ignored when not applicable 🤔

Reply:
I assume we probably want to move the search/filter implementation for the existing use of the ServiceResourceFE model in the API delegation UI also to use the same implementation in BFF later anyway, and support paging there as well.

Just a thought, but it's not critical at this point. Can be cleaned up when rewriting to support it for APIs later.

Reply:
I've already changed it 😛